### PR TITLE
chore: merge upstream v0.34.14

### PR DESCRIPTION
**Pull request by michaelfig**
## Description

Merge in latest upstream v0.34.

Also, fix a replay race in the `committingClient`.  This has `BeginBlock` mark the client as initialized when there is an appHash present.

______

For contributor use:

- [ ] Wrote tests
- [ ] Updated CHANGELOG_PENDING.md
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: abci/client/client.go [hunk 1]
```diff
@@ -81,7 +81,7 @@ type ReqRes struct {
 	*sync.WaitGroup
 	*types.Response // Not set atomically, so be sure to use WaitGroup.
 
-	mtx  tmsync.RWMutex
```
Comment (JimLarson) [inline abci/client/client.go:84]: Please confirm that this is a deliberate revert of the changes on Aug 30.

**Review by JimLarson — APPROVED** [review]
On the understanding that we're intentionally rolling back some of the concurrency changes we made around Aug 30, approved.

File: abci/client/committing_client.go [hunk 2]
```diff
@@ -160,6 +160,13 @@ func (app *committingClient) InitChainAsync(req types.RequestInitChain) *ReqRes
 }
 
 func (app *committingClient) BeginBlockAsync(req types.RequestBeginBlock) *ReqRes {
+	if len(req.Header.AppHash) > 0 && !app.mtx.IsInitialized() {
+		// We already have some state, so mark as initialized.
+		app.mtx.Lock()
+		app.mtx.Initialize()
+		app.mtx.Unlock()
```
Comment (JimLarson) [inline abci/client/committing_client.go:167]: Consider making this `defer app.mtx.Unlock()` and move it up right after the `app.mtx.Lock()`, as this is more idiomatic and handles the case where `app.mtx.Initialize()` misbehaves.  Here and below.

Comment (JimLarson) [reply abci/client/committing_client.go:167]: Nevermind - I was imagining this was followed by a `return` statement, but it isn't.

File: abci/client/client.go (hunk 1 above)
Comment (michaelfig) [reply abci/client/client.go:84]: It is, but do note we're not using this code path in the Agoric SDK.

**Pull request merged**